packages/host/tests/unit/garbage-collection-test.ts

```diff
@@ -658,4 +660,47 @@ module('Unit | identity-context garbage collection', function (hooks) {
       'card instance is returned',
     );
   });
+
+  test('can handle encountering a NotLoaded error in an instance during garbage collection', async function (assert) {
+    let { identityContext } = await setupTest();
+    let doc: LooseSingleCardDocument = {
+      data: {
+        id: `${testRealmURL}wu`,
+        type: 'card',
+        attributes: {
+          name: 'Wu',
+        },
+        relationships: {
+          bestFriend: {
+            links: { self: `${testRealmURL}not-loaded` },
+            // this is what a NotLoaded error looks like: a relationship data.id
+            // that has no associated included resource
+            data: {
+              id: `${testRealmURL}not-loaded`,
+              type: 'card',
+            },
+          },
+        },
+        meta: {
+          adoptsFrom: {
+            module: `${testRealmURL}test-cards`,
+            name: 'Person',
+          },
+        },
+      },
+    };
```

Comment:
@tintinthong FYI this is what a serialization that has a NotLoadedError looks like